src/daemon/bitcoind/actions.rs

```diff
@@ -601,16 +546,98 @@ pub fn bitcoind_main_loop(
             continue;
         }
 
-        // When bitcoind isn't synced, poll each 30s
         if let Some(last_poll) = last_poll {
             if now.duration_since(last_poll) < poll_interval {
-                thread::sleep(poll_interval - now.duration_since(last_poll));
+                thread::sleep(Duration::from_millis(500));
                 continue;
             }
         }
 
         last_poll = Some(now);
-        update_deposits(&mut revaultd, &bitcoind, &mut deposits_cache)?;
-        update_tip(&mut revaultd, &bitcoind)?;
+        update_deposits(
+            &mut revaultd,
+            &bitcoind.read().unwrap(),
+            &mut deposits_cache,
+        )?;
+        update_tip(&mut revaultd, &bitcoind.read().unwrap())?;
     }
+
+    Ok(())
+}
+
+fn wallet_transaction(bitcoind: &BitcoinD, txid: Txid) -> Option<WalletTransaction> {
+    let res = bitcoind.get_wallet_transaction(txid);
+    if let Ok((hex, blockheight, received_time)) = res {
+        Some(WalletTransaction {
+            hex,
+            blockheight,
+            received_time,
+        })
+    } else {
+        log::trace!(
+            "Got '{:?}' from bitcoind when requesting wallet transaction '{}'",
+            res,
+            txid
+        );
+        None
+    }
+}
+
+/// The bitcoind event loop.
+/// Listens for bitcoind requests (wallet / chain) and poll bitcoind every 30 seconds,
+/// updating our state accordingly.
+pub fn bitcoind_main_loop(
+    rx: Receiver<BitcoindMessageOut>,
+    revaultd: Arc<RwLock<RevaultD>>,
+    bitcoind: Arc<RwLock<BitcoinD>>,
+) -> Result<(), BitcoindError> {
+    // The verification progress announced by bitcoind *at startup* thus won't be updated
+    // after startup check. Should be *exactly* 1.0 when synced, but hey, floats so we are
+    // careful.
+    let sync_progress = Arc::new(RwLock::new(0.0f64));
+    // Used to shutdown the poller thread
+    let shutdown = Arc::new(AtomicBool::new(false));
+
+    // We use a thread to 1) wait for bitcoind to be synced 2) poll listunspent
+    let pol_revaultd = revaultd.clone();
+    let pol_bitcoind = bitcoind.clone();
+    let pol_sp = sync_progress.clone();
+    let pol_sd = shutdown.clone();
+    let poller_thread =
+        std::thread::spawn(move || poller_main(pol_revaultd, pol_bitcoind, pol_sp, pol_sd));
```

Comment:
maybe no make it clear these vars are for the thread
```rust
// We use a thread to 1) wait for bitcoind to be synced 2) poll listunspent
    let poller_thread = std::thread::spawn({
        let pol_revaultd = revaultd.clone();
        let pol_bitcoind = bitcoind.clone();
        let pol_sp = sync_progress.clone();
        let pol_sd = shutdown.clone();
        move || poller_main(pol_revaultd, pol_bitcoind, pol_sp, pol_sd)
    });
```

Reply:
oO i can just use the variables in the closue if i move the `move` ? Thanks!

Reply:
This made it much cleaner, applied:
```rust
diff --git a/src/daemon/bitcoind/actions.rs b/src/daemon/bitcoind/actions.rs
index 4e7dffc..9d5055a 100644
--- a/src/daemon/bitcoind/actions.rs
+++ b/src/daemon/bitcoind/actions.rs
@@ -599,12 +599,13 @@ pub fn bitcoind_main_loop(
     let shutdown = Arc::new(AtomicBool::new(false));
 
     // We use a thread to 1) wait for bitcoind to be synced 2) poll listunspent
-    let pol_revaultd = revaultd.clone();
-    let pol_bitcoind = bitcoind.clone();
-    let pol_sp = sync_progress.clone();
-    let pol_sd = shutdown.clone();
-    let poller_thread =
-        std::thread::spawn(move || poller_main(pol_revaultd, pol_bitcoind, pol_sp, pol_sd));
+    let poller_thread = std::thread::spawn({
+        let _revaultd = revaultd.clone();
+        let _bitcoind = bitcoind.clone();
+        let _sync_progress = sync_progress.clone();
+        let _shutdown = shutdown.clone();
+        move || poller_main(_revaultd, _bitcoind, _sync_progress, _shutdown)
+    });
 
     for msg in rx {
         match msg {
```

Will use that from now on!